### DISCUSSION TITLE
How about type hints?

### DISCUSSION THREAD

**Discussion by demberto** [r0x0r/pywebview#1099] [Q&A]
@r0x0r would you accept a PR for this?

**Comment by r0x0r** [comment]
I am not a big fan of typing, but if you are ready to do this, then go ahead.
One request though, when submitting a PR, please make sure that it does not have any loose ends, as I don't want to spend time on fixing those. 
Typing platform specific code is a tedious task, so it can be left out for now.

**Reply by demberto** [reply]
Great :) I will give it a go then.

**Reply by demberto** [reply]
@r0x0r Few questions:
* What versions of Python do you exactly support? (I see a lot of Py 2 era code)
* Should I inline the type hints?
* Do you still plan to keep Python 3.6 supported? If no, I can use `from __future__ import annotations` (need Python 3.7+), which results in massively improved type hints.

**Comment by r0x0r** [comment]
3.6 is the minimum for now. I have no particular reason to keep it at 3.6. If 3.7 makes code cleaner, then go for it.
Support for Python 2 was dropped with 3.0, but code has never been cleaned up properly. Feel free to remove Python 2 related code.

**Reply by demberto** [reply]
Alright ✅

**Reply by demberto** [reply]
What line length, formatter do you use? I can configure my environment accordingly.
